packages/browser/src/index.ts

```diff
@@ -29,17 +33,54 @@ export type AccessToken = {
   expiresAt: number; // Unix Timestamp in seconds
 };
 
+export const LogtoSignInSessionItemSchema = type({
+  redirectUri: string(),
+  codeVerifier: string(),
+  state: string(),
+});
+
+export type LogtoSignInSessionItem = Infer<typeof LogtoSignInSessionItemSchema>;
+
 export default class LogtoClient {
   protected accessTokenMap = new Map<string, AccessToken>();
   protected refreshToken?: string;
   protected idToken?: string;
   protected logtoConfig: LogtoConfig;
   protected oidcConfig?: OidcConfigResponse;
 
+  private readonly _signInSession: Optional<LogtoSignInSessionItem>;
+
   constructor(logtoConfig: LogtoConfig) {
     this.logtoConfig = logtoConfig;
   }
 
+  protected get signInSession(): LogtoSignInSessionItem {
+    const logtoKey = getLogtoKey(this.logtoConfig.clientId);
+    const jsonItem = sessionStorage.getItem(logtoKey);
+
+    if (!jsonItem) {
+      throw new LogtoClientError('sign_in_session.not_found');
+    }
+
+    try {
+      const item: unknown = JSON.parse(jsonItem);
+
+      return create(item, LogtoSignInSessionItemSchema);
+    } catch (error: unknown) {
+      if (error instanceof SyntaxError) {
```

Comment:
Should we wrap all errors into LogtoClientError for a better dev experience?

Reply:
I intended to prompt the JSON.parse failure due to the invalid sign-in session.

Assume that the sign-up session is all under our control, the try-catch is not necessary enough here.

Removed now. PTAL.